### ISSUE TITLE
applyPatch missing null in return type

### ISSUE THREAD

**Issue by matthew-nichols-westtel** [chbrown/rfc6902#75]
Version 4.0.0 on NPM (the latest), the type for applyPatch appears to be wrong (inside `index.d.ts`):

```typescript
export declare function applyPatch(object: any, patch: Operation[]): (import("./patch").MissingError | import("./patch").TestError | import("./patch").InvalidOperationError)[];
```

I'm pretty sure it should be

```typescript
export declare function applyPatch(object: any, patch: Operation[]): (import("./patch").MissingError | import("./patch").TestError | import("./patch").InvalidOperationError | null)[];
```

Aka the return value is missing a `| null`.

**Comment by chbrown** [comment]
You're 100% correct, but this appears to be a TypeScript bug (I'm using v4.0.3) — for some reason when `tsc` is inferring the return type of `applyPatch`, it's dropping the `null`. The return type of `applyPatch` is pretty straightforwardly an array of the return type of `patch.apply`, which _is_ explicitly specified, at https://github.com/chbrown/rfc6902/blob/v4.0.0/patch.ts#L200:
```typescript
export function apply(object: any, operation: Operation): MissingError | InvalidOperationError | TestError | null {
```
Surprisingly, _that_ `null` surfaces in `patch.d.ts`, as it should!

Sure, I could explicitly state the return type of `applyPatch`, but I don't understand why the TypeScript compiler has a problem with this.

Btw is this breaking something for you? What brought it to your attention?

**Comment by matthew-nichols-westtel** [comment]
The combination of the [@typescript-eslint/strict-boolean-expressions](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/strict-boolean-expressions.md) lint and the following pattern:

```typescript
const errors = applyPatch(obj, patches);
for (const e of errors) {
  if (e) {
    throw e;
  }
}
```

causes a lint error at `if (e)` because it thinks the type of `e` is truthy when it's not.

I cloned and took another look and it's not a TypeScript bug. rfc6902 is compiled without the [strict](https://www.typescriptlang.org/tsconfig#strict) option (more specifically [strictNullChecks](https://www.typescriptlang.org/tsconfig#strictNullChecks)), without which TypeScript drops the null. Enabling it causes the output to have the null.

**Comment by chbrown** [comment]
Okay, not a bug, but a little unintuitive 😜 

Switched to `strict` mode, which only raised a few extra quibbles... easy to fix.

Released as v4.0.1 just now 👍
